[Documentation] fixes for configauth module's README (#4439)
open-telemetry/opentelemetry-collector: config/configauth/README.md

# Authentication configuration

> This module defines necessary interfaces to implement 

Reviewer: ```suggestion
This module defines necessary interfaces to implement server and client type authenticators:
```

> - Server type authenticators for gRPC and HTTP receivers to perform authentication for requests and/or RPCs. Each server type is responsible for getting the request/RPC metadata and passing down to the authenticator.

Reviewer: ```suggestion
- Server type authenticators perform authentication for incoming HTTP/gRPC requests and are typically used in receivers.
```

Not sure what you meant by the last statement.

> - Client type authenticators for gRPC and HTTP exporters to perform client side authentication. 

Reviewer: ```suggestion
- Client type authenticators perform client-side authentication for outgoing HTTP/gRPC requests and are typically used in exporters.
```

> Few such authenticators that are implemented currently:

Reviewer: Those are not "a few authenticators", those are all known authenticators :-) Perhaps something like:

```suggestion
The currently known authenticators are:
```

    issuer_url: http://localhost:8080/auth/realms/opentelemetry
    audience: account

>   oauth2client:

Reviewer: Did you try those examples?

Author: yes @jpkrohling  this is from the E2E test we carried out  https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/3848#issue-926422174 
Works fine.

        auth:
          ## oidc is the extension name to use as the authenticator for this receiver
          authenticator: oidc

  otlphttp/withauth:
    endpoint: http://localhost:9000
    auth:
      authenticator: oauth2client

```

## Creating an authenticator

> New server side authenticators can be added by creating a new extension that also implements the `configauth.ServerAuthenticator` extension and similarly a client side authenticator can be added by implementing `configauth.ClientAuthenticator` interface.

Reviewer: ```suggestion
New authenticators can be added by creating a new extension that also implements the appropriate interface (`configauth.ServerAuthenticator` or `configauth.ClientAuthenticator`).
```

> Generic authenticators that may be used by a good number of users might be accepted as part of the core distribution, or as part of the contrib distribution. If you have interest in contributing one authenticator, open an issue with your proposal. For other cases, you'll need to include your custom authenticator as part of your custom OpenTelemetry Collector, perhaps being built using the [OpenTelemetry Collector Builder](https://github.com/open-telemetry/opentelemetry-collector-builder).

Reviewer: ```suggestion
Generic authenticators that may be used by a good number of users might be accepted as part of the contrib distribution. If you have an interest in contributing an authenticator, open an issue with your proposal. For other cases, you'll need to include your custom authenticator as part of your custom OpenTelemetry Collector, perhaps being built using the [OpenTelemetry Collector Builder](https://github.com/open-telemetry/opentelemetry-collector/tree/main/cmd/builder).
```
